compiler/modules/types/list.bal

```diff
@@ -22,6 +22,21 @@ type ListConjunction record {|
     ListConjunction? next;
 |};
 
+public type ListMemberTypes [Range[], SemType[]];
```

Comment:
We should maintain the invariant that the SemType array here does not have entries == NEVER.  Please add a comment to that effect.